index.js

```diff
@@ -470,21 +494,45 @@ const createFocusTrap = function (elements, userOptions) {
         return this;
       }
 
-      updateTabbableNodes();
-
       state.active = true;
       state.paused = false;
       state.nodeFocusedBeforeActivation = doc.activeElement;
 
-      const onActivate =
-        activateOptions && activateOptions.onActivate
-          ? activateOptions.onActivate
-          : config.onActivate;
-      if (onActivate) {
-        onActivate();
```

Comment:
I think this should remain here. After `onActivate()` has been called, then the remaining things to call are `updateTabbableNodes()` and `addListeners()` -- and now the new `onPostActivate()` -- and these are the things we want to optionally delay with `checkCanActivate()`. As it is, with your changes, we're still immediately calling `onActivate()` synchronously, we're just doing it twice, one in the IF and one in the ELSE.

Reply:
I think this is resolved now.